ReplayKitExample/ReplayKitExample/ReplayKitVideoSource.swift

```diff
@@ -63,74 +64,64 @@ class ReplayKitVideoSource: NSObject, TVIVideoCapturer {
             screenSize.width *= UIScreen.main.nativeScale
             screenSize.height *= UIScreen.main.nativeScale
-            let format = TVIVideoFormat()
-            format.pixelFormat = TVIPixelFormat.formatYUV420BiPlanarFullRange
-            format.frameRate = UInt(ReplayKitVideoSource.kFormatFrameRate)
-            format.dimensions = CMVideoDimensions(width: Int32(screenSize.width), height: Int32(screenSize.height))
-
-            // We will downscale buffers to a 640x640 box, if requested.
-            let downscaledFormat = TVIVideoFormat()
-            downscaledFormat.frameRate = UInt(ReplayKitVideoSource.kFormatFrameRate)
-            downscaledFormat.pixelFormat = TVIPixelFormat.formatYUV420BiPlanarFullRange
-            downscaledFormat.dimensions = CMVideoDimensions(width: Int32(ReplayKitVideoSource.kDownScaledMaxWidthOrHeight), height: Int32(ReplayKitVideoSource.kDownScaledMaxWidthOrHeight))
-
-            return [format, downscaledFormat]
+            outputFormat.pixelFormat = TVIPixelFormat.formatYUV420BiPlanarFullRange
+            outputFormat.frameRate = UInt(ReplayKitVideoSource.kFormatFrameRate)
+            outputFormat.dimensions = CMVideoDimensions(width: Int32(screenSize.width), height: Int32(screenSize.height))
         }
+
+        super.init()
+        requestOutputFormat(outputFormat)
     }
 
-    deinit {
-        freeScalingBuffers()
+    public var isScreencast: Bool {
+        get {
+            return screencastUsage
+        }
     }
 
-    func startCapture(_ format: TVIVideoFormat, consumer: TVIVideoCaptureConsumer) {
-        captureConsumer = consumer
+    func requestOutputFormat(_ outputFormat: TVIVideoFormat) {
+        downscaleBuffers = (outputFormat.dimensions.width == ReplayKitVideoSource.kDownScaledMaxWidthOrHeight &&
+                            outputFormat.dimensions.height == ReplayKitVideoSource.kDownScaledMaxWidthOrHeight)
+        videoFormat = outputFormat
 
-        if (format.dimensions.width == ReplayKitVideoSource.kDownScaledMaxWidthOrHeight &&
-            format.dimensions.height == ReplayKitVideoSource.kDownScaledMaxWidthOrHeight) {
-            downscaleBuffers = true
+        if let sink = sink {
+            sink.onVideoFormatRequest(videoFormat)
         }
-
-        print("Start capturing with format:", format)
     }
 
-    func stopCapture() {
-        print("Stop capturing.")
-
+    deinit {
         // Perform teardown and free memory on the video queue to ensure that the resources will not be resurrected.
         if let captureQueue = self.videoQueue {
             captureQueue.sync {
-                self.captureConsumer = nil
                 self.timerSource?.cancel()
                 self.timerSource = nil
-                self.freeScalingBuffers()
             }
-        } else {
-            // Fallback in case we never captured any frames yet.
-            captureConsumer = nil
         }
+
+        freeScalingBuffers()
```

Comment:
This class should no longer need any scaling buffers, or any downscaling code at all since this is now offered "for free" so to speak by TVIVideoSource.